### PR TITLE
feat: added deepgram for transcription

### LINKED ISSUE #96: feat: use deepgram for transcription

### PR DESCRIPTION
**Pull request by masterchief164**
This PR fixes #96.

### CODE REVIEW

File: app/application.py [hunk 1]
```diff
@@ -193,7 +193,7 @@ def decimal_to_sexagesimal(dec):
     return f'{hrs}:{minu}:{sec}'
 
 
-def process_mp3(filename, model):
```
Comment (adamjonas) [inline app/application.py:186]: I'd suggest reverting this commit

**Review by adamjonas — COMMENTED** [review]
In testing it out, whisper is much more accurate than the default nova model. I'd suggest leaving the current whisper code as is and adding a flag to use deepgram but be able to pass it various models.

One of the problems is that the chapters breaking up the timestamps. Instead of breaking up the file, I'd suggest adding the headers by locating the timestamp and keeping the file intact. That might be difficult without granular timestamps like deepgram provides, but I do think it can be done.

Additionally, the default should be with the speaker diarization off and turn it on via flag and/or adding multiple speakers to the speakers metadata.

**Comment by masterchief164** [comment]
> One of the problems is that the chapters breaking up the timestamps. Instead of breaking up the file, I'd suggest adding the headers by locating the timestamp and keeping the file intact. That might be difficult without granular timestamps like deepgram provides, but it should be possible even with the whisper srt format. If we keep the file intact, we also should get summarization via deepgram which can close #38.

Could you please provide an example of the first part.

**Comment by adamjonas** [comment]
https://github.com/adamjonas/bitcointranscripts/pull/113/files

**Comment by adamjonas** [comment]
testing the chapters

` tstbtc '1Z7GjXgdUy4' test-dir -p -D -C` produces https://github.com/adamjonas/bitcointranscripts/pull/146/files and https://github.com/adamjonas/bitcointranscripts/pull/145/files, which both cut off before transcribing the last chapter of the video.

**Comment by masterchief164** [comment]
This PR also fixes #38

**Pull request merged**